Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -118,15 +118,126 @@ public static bool IsBase64EncodingOfPrintableCharacters(string secret)
             return true;
         }
 
-        public static bool ContainsDigitAndChar(string matchedPattern)
+        /// <summary>
+        /// Determines whether the input string contains both digit and letter.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <returns>true if the input string contains both digit and letter; otherwise, false.</returns>
+        public static bool ContainsDigitAndChar(string text)
+        {
+            // TODO, change the function name as ContainsDigitAndLetter and also fixed the 44 references
+            return ContainAtLeastNDigits(text, 1) && ContainAtLeastNLetters(text, 1);
+        }
+
+        /// <summary>
+        /// Determines whether the input string contains both lowercase and uppercase letter.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <returns>true if the input string contains both lowercase and uppercase letter; otherwise, false.</returns>
+        public static bool ContainLowercaseAndUppercaseLetter(string text)
+        {
+            return ContainAtLeastNLowercaseLetters(text, 1) && ContainAtLeastNUppercaseLetters(text, 1);
+        }
+
+        /// <summary>
+        /// Determines whether the input string contains at least n digits.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <param name="n">The number of digits required.</param>
+        /// <returns>true if the input string contains at least n digits; otherwise, false.</returns>
+        public static bool ContainAtLeastNDigits(string text, int n)
```

Comment:
@LingZhou-gh Consider renaming the variable `n` since it is too generic. How about `lowerBound`?